File: go_gin_sqlx/internal/model/bookModel.go
package model

import "time"

type Book struct {
	Id        uint      `json:"id" db:"id"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
	ISBN      string    `json:"isbn" db:"isbn"`
	Name      string    `json:"name" db:"name"`
	AuthorId  uint      `json:"author_id" db:"author_id"`
	CreatedBy uint      `json:"created_by" db:"created_by"`
}

type BookAuthorCreator struct {
	Book   `db:"book"`
	Author `json:"author" db:"author"`
	User   `json:"creator" db:"user"`
}

File: spring_boot/src/main/java/books/books_anything/models/BookModel.java
package books.books_anything.models;

import java.util.Date;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.FetchType;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data
@NoArgsConstructor
@RequiredArgsConstructor
@Entity
public class BookModel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @CreationTimestamp
    private Date createdAt;

    @UpdateTimestamp
    private Date updatedAt;

    @NonNull
    private String isbn;

    @NonNull
    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "author_id")
    private AuthorModel author;
}
